src/CodeGen_Internal.cpp

```diff
@@ -585,43 +585,41 @@ Expr lower_signed_shift_left(const Expr &a, const Expr &b) {
     internal_assert(b.type().is_int());
     const int64_t *const_int_b = as_const_int(b);
     if (const_int_b) {
-        Type t = UInt(a.type().bits(), a.type().lanes());
         Expr val;
         const uint64_t b_unsigned = std::abs(*const_int_b);
         if (*const_int_b >= 0) {
-            val = a << make_const(t, b_unsigned);
+            val = a << make_const(UInt(a.type().bits()), b_unsigned);
```

Comment:
Would it help readability to add `.as_uint()` for these cases? (We have `with_code()` but it's not really any terser)

Reply:
Note this is also dropping the number of lanes too.